urns/urns.go

```diff
@@ -3,125 +3,33 @@ package urns
-	ViberScheme      string = "viber"
-	VKScheme         string = "vk"
-	WebChatScheme    string = "webchat"
-	WeChatScheme     string = "wechat"
-	WhatsAppScheme   string = "whatsapp"
-
-	// FacebookRefPrefix is prefix used for facebook referral URNs
-	FacebookRefPrefix string = "ref:"
+	maxPathLength    = 255
+	maxDisplayLength = 255
 )
 
-// ValidSchemes is the set of URN schemes understood by this library
-var ValidSchemes = map[string]bool{
-	DiscordScheme:    true,
-	EmailScheme:      true,
-	ExternalScheme:   true,
-	FacebookScheme:   true,
-	FCMScheme:        true,
-	FreshChatScheme:  true,
-	InstagramScheme:  true,
-	JiochatScheme:    true,
-	LineScheme:       true,
-	RocketChatScheme: true,
-	SlackScheme:      true,
-	TeamsScheme:      true,
-	TelegramScheme:   true,
-	TelScheme:        true,
-	TwitterIDScheme:  true,
-	TwitterScheme:    true,
-	ViberScheme:      true,
-	VKScheme:         true,
-	WebChatScheme:    true,
-	WeChatScheme:     true,
-	WhatsAppScheme:   true,
-}
-
 // IsValidScheme checks whether the provided scheme is valid
 func IsValidScheme(scheme string) bool {
-	_, valid := ValidSchemes[scheme]
+	_, valid := schemes[scheme]
 	return valid
 }
 
-var nonTelCharsRegex = regexp.MustCompile(`[^0-9a-z]`)
-var telRegex = regexp.MustCompile(`^\+?[a-zA-Z0-9]{1,64}$`)
-var twitterHandleRegex = regexp.MustCompile(`^[a-zA-Z0-9_]{1,15}$`)
-var emailRegex = regexp.MustCompile(`^[^\s@]+@[^\s@]+$`)
-var viberRegex = regexp.MustCompile(`^[a-zA-Z0-9_=/+]{1,24}$`)
-var lineRegex = regexp.MustCompile(`^[a-zA-Z0-9_]{1,36}$`)
-var allDigitsRegex = regexp.MustCompile(`^[0-9]+$`)
-var freshchatRegex = regexp.MustCompile(`^[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}/[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$`)
-var webchatRegex = regexp.MustCompile(`^[a-zA-Z0-9]{24}(:[^\s@]+@[^\s@]+)?$`)
+// Schemes returns the valid URN schemes
+func Schemes() []string {
+	return schemePrefixes
+}
 
 // URN represents a Universal Resource Name, we use this for contact identifiers like phone numbers etc..
 type URN string
 
-// NewTelURNForCountry returns a URN for the passed in telephone number and country code ("US")
-func NewTelURNForCountry(number string, country string) (URN, error) {
-	return NewURNFromParts(TelScheme, normalizeNumber(number, country), "", "")
-}
-
-// NewTelegramURN returns a URN for the passed in telegram identifier
-func NewTelegramURN(identifier int64, display string) (URN, error) {
-	return NewURNFromParts(TelegramScheme, strconv.FormatInt(identifier, 10), "", display)
-}
-
-// NewWhatsAppURN returns a URN for the passed in whatsapp identifier
-func NewWhatsAppURN(identifier string) (URN, error) {
-	return NewURNFromParts(WhatsAppScheme, identifier, "", "")
-}
-
-// NewFirebaseURN returns a URN for the passed in firebase identifier
-func NewFirebaseURN(identifier string) (URN, error) {
-	return NewURNFromParts(FCMScheme, identifier, "", "")
-}
```

Comment:
we only created constructors for some schemes.. and we don't need them

Reply:
Those were used in courier but we can update courier to use `NewURNFromParts`

Reply:
Or as it is most common to not have a `query` and `display` for the URN can we have a short cut that we pass the `scheme` type and the `path` and that set the `query` and `display` to `""`?

Reply:
Have added that as `urns.New(scheme, path)`